Keep docks menu open when opening dock

kitbdev opened the pull request:
- see this comment https://github.com/godotengine/godot/pull/89017#issuecomment-2089450840

If opening a dock, keep the menu open so more docks can be opened.
When focusing a dock, it closes.

![godot dockmenu stay open](https://github.com/godotengine/godot/assets/10054226/f6136d28-c581-4968-a923-8b2b13ab369c)

kitbdev committed:
Keep docks menu open when opening dock

KoBeWi commented on editor/editor_dock_manager.cpp:
```diff
@@ -180,7 +180,17 @@ void EditorDockManager::_update_docks_menu() {
+	ERR_FAIL_NULL(dock);
+	ERR_FAIL_COND_MSG(!all_docks.has(dock), vformat("Menu option for unknown dock '%s'.", dock->get_name()));
+	if (all_docks[dock].enabled && all_docks[dock].open) {
+		docks_menu->hide();
```
Hiding parent menu is enough. I think you can assume it exists and do ERR_FAIL_NULL if it doesn't for some reason.

KoBeWi reviewed (approved):
I wonder if it wouldn't be better to make `focus_dock()` return a `bool` instead of partially duplicating it, but it's fine this way too.

kitbdev commented:
Rebased and updated.
It's not really duplicating part of focus_dock, just changing behavior based on current open/closed state, I'd say.
Also if we do want to change to open/close and checkboxes like mentioned in RC, then manually closing the parent menu like this won't be necessary.

akien-mga commented:
Thanks!
